src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -43,8 +46,24 @@ NamesAndTypesList StorageSystemZooKeeper::getNamesAndTypes()
     };
 }
 
+using Paths = Strings;
 
-static bool extractPathImpl(const IAST & elem, String & res, const Context & context)
+static String pathCorrected(const String & path)
+{
+    String path_corrected;
+    /// path should starts with '/', otherwise ZBADARGUMENTS will be thrown in
+    /// ZooKeeper::sendThread and the session will fail.
```

Comment:
Btw, it looks like a bug in ZooKeeper client or TestKeeper, incorrect path should not break session. Moreover, the following query hangs on `Poco::Event` inside ZooKeeper client:
```
:) system drop replica 'test' from zkpath 'session/expired'
```